katharsis-client/src/main/java/io/katharsis/client/internal/DefaultQueryParamsSerializer.java

```diff
@@ -0,0 +1,119 @@
+		TypedParams<SortingParams> sorting = queryParams.getSorting();
+		if (sorting != null) {
+			Map<String, SortingParams> sortingParamsMap = sorting.getParams();
+			for (Map.Entry<String, SortingParams> entry : sortingParamsMap.entrySet()) {
+				String type = entry.getKey();
+				Map<String, RestrictedSortingValues> params = entry.getValue().getParams();
+				for (Entry<String, RestrictedSortingValues> param : params.entrySet()) {
+					String paramName = RestrictedQueryParamsMembers.sort.name() + "[" + type + "][" + param.getKey() + "]";
+					map.put(paramName, param.getValue().toString());
+				}
+			}
+		}
+		return map;
+	}
+
+	@Override
+	public Map<String, Set<String>> serializeIncludedFields(QueryParams queryParams) {
+		Map<String, Set<String>> map = new HashMap<String, Set<String>>();
+		TypedParams<IncludedFieldsParams> includedFields = queryParams.getIncludedFields();
+		if (includedFields != null) {
+			Map<String, IncludedFieldsParams> includedFieldsParamsMap = includedFields.getParams();
+			for (Entry<String, IncludedFieldsParams> entry : includedFieldsParamsMap.entrySet()) {
+				String type = entry.getKey();
+				Set<String> values = entry.getValue().getParams();
+				String paramName = RestrictedQueryParamsMembers.fields.name() + "[" + type + "]";
+				map.put(paramName, values);
+			}
+		}
+		return map;
+	}
+
+	@Override
+	public Map<String, Set<String>> serializeIncludedRelations(QueryParams queryParams) {
+		Map<String, Set<String>> map = new HashMap<String, Set<String>>();
+		TypedParams<IncludedRelationsParams> includedRelations = queryParams.getIncludedRelations();
+		if (includedRelations != null) {
+			Map<String, IncludedRelationsParams> includedRelationsParamsMap = includedRelations.getParams();
+			for (Entry<String, IncludedRelationsParams> entry : includedRelationsParamsMap.entrySet()) {
+				String type = entry.getKey();
+				Set<Inclusion> inclusions = entry.getValue().getParams();
+
+				Set<String> strInclusions = new HashSet<String>();
+				for (Inclusion inclusion : inclusions) {
+					strInclusions.add(inclusion.getPath());
+				}
+
+				String paramName = RestrictedQueryParamsMembers.include.name() + "[" + type + "]";
+				map.put(paramName, strInclusions);
+			}
+		}
+		return map;
+	}
+
+	@Override
+	public Map<String, String> serializePagination(QueryParams queryParams) {
+		Map<String, String> map = new HashMap<String, String>();
+		Map<RestrictedPaginationKeys, Integer> pagination = queryParams.getPagination();
+		if (pagination != null) {
+			for (Entry<RestrictedPaginationKeys, Integer> entry : pagination.entrySet()) {
+				String paramName = RestrictedQueryParamsMembers.page.name() + "[" + entry.getKey().name() + "]";
+				map.put(paramName, entry.getValue().toString());
+			}
+		}
+		return map;
+	}
+
+	@Override
+	public Map<String, Set<String>> serializeGrouping(QueryParams queryParams) {
+		Map<String, Set<String>> map = new HashMap<String, Set<String>>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "map". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1488)